x-pack/plugins/security_solution/server/lib/detection_engine/rules/get_export_by_object_ids.ts

```diff
@@ -48,33 +49,48 @@ export const getRulesFromObjects = async (
   alertsClient: AlertsClient,
   objects: Array<{ rule_id: string }>
 ): Promise<RulesErrors> => {
-  const alertsAndErrors = await Promise.all(
-    objects.reduce<Array<Promise<ExportRules>>>((accumPromise, object) => {
-      const exportWorkerPromise = new Promise<ExportRules>(async (resolve) => {
-        try {
-          const rule = await readRules({ alertsClient, ruleId: object.rule_id, id: undefined });
-          if (rule != null && isAlertType(rule) && rule.params.immutable !== true) {
-            const transformedRule = transformAlertToRule(rule);
-            resolve({
-              statusCode: 200,
-              rule: transformedRule,
-            });
-          } else {
-            resolve({
-              statusCode: 404,
-              missingRuleId: { rule_id: object.rule_id },
-            });
-          }
-        } catch {
-          resolve({
-            statusCode: 404,
-            missingRuleId: { rule_id: object.rule_id },
-          });
-        }
-      });
-      return [...accumPromise, exportWorkerPromise];
-    }, [])
-  );
+  // If we put more than 1024 ids in one block like "alert.attributes.tags: (id1 OR id2 OR ... OR id1100)"
+  // then the KQL -> ES DSL query generator still puts them all in the same "should" array, but ES defaults
+  // to limiting the length of "should" arrays to 1024. By chunking the array into blocks of 1024 ids,
+  // we can force the KQL -> ES DSL query generator into grouping them in blocks of 1024.
+  // The generated KQL query here looks like
+  // "alert.attributes.tags: (id1 OR id2 OR ... OR id1024) OR alert.attributes.tags: (...) ..."
+  const chunkedObjects = chunk(objects, 1024);
```

Comment:
I manually tested this chunking and filter logic to ensure that we can export more than 1024 rules. I wrote an integration test to cover it as well, but creating 1000+ rules is a bottleneck as it can take 1 min or more to create that many rules. The test code is in the PR description, so if we can remove that bottleneck then adding the test back is simple. An alternative would be to try to use an esArchive to import 1000+ rules and bypass the rule creation process, but that couples our test to the alerting framework and doesn't seem worth the additional maintenance burden.